go-controller/pkg/cluster/gateway_init.go

```diff
@@ -30,9 +30,33 @@ func bridgedGatewayNodeSetup(nodeName, bridgeInterface string) (string, string,
 	}
 
 	// ovn-bridge-mappings maps a physical network name to a local ovs bridge
-	// that provides connectivity to that network.
+	// that provides connectivity to that network. It is in the form of physnet1:br1,physnet2:br2.
+	// Note that there may be multiple ovs bridge mappings, be sure not to override
+	// the mappings for the other physical network
+	stdout, stderr, err = util.RunOVSVsctl("--if-exists", "get", "Open_vSwitch", ".",
+		"external_ids:ovn-bridge-mappings")
+	if err != nil {
+		return "", "", fmt.Errorf("Failed to get ovn-bridge-mappings stderr:%s (%v)", stderr, err)
+	}
+	// skip the existing mapping setting for the specified physicalNetworkName
+	mapString := ""
+	bridge_mappings := strings.Split(stdout, ",")
+	for _, bridge_mapping := range bridge_mappings {
+		m := strings.Split(bridge_mapping, ":")
+		if network := m[0]; network != physicalNetworkName {
+			if len(mapString) != 0 {
+				mapString += ","
+			}
+			mapString += bridge_mapping
+		}
+	}
+	if len(mapString) != 0 {
+		mapString += ","
+	}
+	mapString += physicalNetworkName+":"+bridgeInterface
+
```

Comment:
Can we siimplify lines L42-56 to do this?
```
br_mapping := physicalNetworkName+":"+bridgeInterface
ovn_bridge_mappings := util.RunOVSVsctl(get_external_ids_ovn_bridge_mappings)
if strings.Index(ovn_bridge_mappings, br_mapping) != -1 {
    if (len(ovn_bridge_mappings) != 0)
        ovn_bridge_mappings += ","
    ovn_bridge_mappings += br_mapping
}
```

Reply:
first, you meant if strings.Index(ovn_bridge_mappings, br_mapping) == -1, right? Second, what if the physicalNetworkName is there but with incorrect bridgeinterfaceName? I think that is very possible case if ovn-bridge-mappings are not cleaned up.

Reply:
@cathy-zhou 

Yes, my bad .. i meant `strings.Index(ovn_bridge_mappings, br_mapping) == -1`.

I highly doubt that would be the case. However, let us keep what you have for now.